### PR TITLE
Fixes #4341: Adds a new organization taxonomy that is labeled empty.

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by skottler** [comment]
:+1:

**Comment by domcleal** [comment]
We added factory girl as a dep a while ago for plugins to start using, I think it'd be better if we added a simple factory for what you want rather than an individual fixture.  It might be less prone to breakage, or being repurposed for another test (a common occurrence).

File: test/test_helper.rb [hunk 1]
```diff
@@ -22,6 +22,9 @@
   require 'capybara/rails'
   require 'factory_girl_rails'
 
+  # Remove previous test log to speed tests up
+  FileUtils.rm("#{Rails.root}/log/test.log")
```
Comment (domcleal) [inline test/test_helper.rb]: Should this be above the initialisation routine, else we might lose messages.

**Comment by ehelms** [comment]
@domcleal Katello tests rely heavily on fixtures and are currently broken without something like this present. Having a factory for this might be good for plugins using factories (we've had issues with factories in the past), but we need this fixture.

**Comment by domcleal** [comment]
But those are katello fixtures right, not fixtures from Foreman's own tests?

**Comment by ehelms** [comment]
@domcleal We are extending existing Foreman objects (like Organization) and need to test our objects with that functionality on top of it. Rails doesn't provide a method to add your own versions of existing fixtures and so we must rely on fixtures defined by Foreman for all Foreman entities (which is admittedly fragile, but the fixture code in Rails is not fun from my experience).

**Comment by domcleal** [comment]
Ouch, that is very fragile, as we do change them on a whim.  Is Katello using fixtures too then, so you need a fixture in Foreman to refer to for associations?  Else starting on the factory route would probably be a much safer idea.

File: test/test_helper.rb (hunk 1 above)
Comment (ehelms) [inline test/test_helper.rb]: Updated.

**Comment by GregSutcliffe** [comment]
The fact that Rails won't let you redefine or add fixtures is _exactly_ why we added FactoryGirl, which will allow it. I'm definitely :+1: for moving to FactoryGirl for that reason alone :)

**Comment by ehelms** [comment]
@domcleal Correct, we have a lot of tests that expect an organization with lots of associations bound to it on our side. The factory route would be time consuming and I'm trying to unblock our core test suite. I agree that we need a better solution at some point - whether we can bend factories to our needs or figure out how to supply our own fixtures for extended objects.

**Comment by domcleal** [comment]
Okay, that makes sense then.  Let's wait for Jenkins.

**Comment by domcleal** [comment]
Merged as 6181dd9 for Foreman 1.5.0, thanks @ehelms.

**Pull request closed without merging**